Source/Core/DolphinWX/LuaScriptFrame.cpp

```diff
@@ -0,0 +1,553 @@
+
+  // Create function map if it doesn't already exist
+  if (!registered_functions)
+  {
+    registered_functions = new std::map<const char*, LuaFunction>;
+
+    registered_functions->insert(std::pair<const char*, LuaFunction>("print", printToTextCtrl));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("frameAdvance", frameAdvance));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("getFrameCount", getFrameCount));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("getMovieLength", getMovieLength));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("softReset", softReset));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("saveState", saveState));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("loadState", loadState));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("setEmulatorSpeed", setEmulatorSpeed));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getAnalog", getAnalog));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setAnalog", setAnalog));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getCStick", getCStick));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setCStick", setCStick));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("setCStickPolar", setCStickPolar));
+    registered_functions->insert(
+        std::pair<const char*, LuaFunction>("setAnalogPolar", setAnalogPolar));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getButtons", getButtons));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setButtons", setButtons));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setDPad", setDPad));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("getTriggers", getTriggers));
+    registered_functions->insert(std::pair<const char*, LuaFunction>("setTriggers", setTriggers));
+  }
+}
+
+LuaScriptFrame::~LuaScriptFrame()
+{
+  // Disconnect Events
+  this->Disconnect(wxID_ANY, wxEVT_COMMAND_MENU_SELECTED,
+                   wxCommandEventHandler(LuaScriptFrame::OnClearClicked));
+  browse_button->Disconnect(wxEVT_COMMAND_BUTTON_CLICKED,
+                            wxCommandEventHandler(LuaScriptFrame::BrowseOnButtonClick), NULL, this);
+  run_button->Disconnect(wxEVT_COMMAND_BUTTON_CLICKED,
+                         wxCommandEventHandler(LuaScriptFrame::RunOnButtonClick), NULL, this);
+  stop_button->Disconnect(wxEVT_COMMAND_BUTTON_CLICKED,
+                          wxCommandEventHandler(LuaScriptFrame::StopOnButtonClick), NULL, this);
+
+  // Stop currently executing Lua thread
+  if (lua_thread != nullptr && lua_thread->IsRunning())
+  {
+    lua_thread->Kill();
+    lua_thread = nullptr;
+  }
+
+  main_frame->m_lua_script_frame = nullptr;

```

Comment:
Please don't do this. If this sort of thing is needed it should be done by sending an event back to the main window, not by directly having a child modify a parent.